### PR TITLE
Add support for ticky ticky counts

### PR DESCRIPTION
**Pull request by nineonine**
This patch add support for ticky ticky count events (which was [added recently](https://gitlab.haskell.org/ghc/ghc/-/merge_requests/3085/diffs#19d24fcbecb583c03fb5bcbb9eed98c81bdd6184))

### CODE REVIEW

**Comment by nineonine** [comment]
cc @bgamari

**Comment by maoe** [comment]
Thank you for your contribution!

I'll take a closer look later but could you please add a test case that contains some ticky-ticky counts? We have [a test suite](https://github.com/haskell/ghc-events/blob/a3ef83e9fa2590f0e79a9ba5a060214869763126/test/TestVersions.hs) which checks that `ghc-events show` output matches the expected output. You can find a list of example eventlogs [here](https://github.com/haskell/ghc-events/blob/master/test/Utils.hs#L3-L23).

**Comment by bgamari** [comment]
Indeed a test would be good but this otherwise looks good to me! Thanks @nineonine.

**Comment by nineonine** [comment]
> Thank you for your contribution!
> 
> I'll take a closer look later but could you please add a test case that contains some ticky-ticky counts? We have [a test suite](https://github.com/haskell/ghc-events/blob/a3ef83e9fa2590f0e79a9ba5a060214869763126/test/TestVersions.hs) which checks that `ghc-events show` output matches the expected output. You can find a list of example eventlogs [here](https://github.com/haskell/ghc-events/blob/master/test/Utils.hs#L3-L23).

For sure! I added a test.

File: test/ticky-ticky.eventlog.reference [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+4407000: cap 0: all caps stopped for GC
+4407000: cap 0: GC stats for heap capset 0: generation 1, 3320 bytes copied, 25256 bytes slop, 942080 bytes fragmentation, 1 par threads, 0 bytes max par copied, 3320 bytes total par copied, 0 bytes balanced par copied
+4407000: cap 0: live data in heap capset 0: 44376 bytes
+4408000: cap 0: size of heap capset 0: 2097152 bytes
+12419000: ticky counter sample 4535063528: entry count: 1, 0 allocs, 0 allocd
+12420000: ticky counter sample 4535063632: entry count: 1, 0 allocs, 0 allocd
+12421000: ticky counter sample 4535063728: entry count: 1, 0 allocs, 0 allocd
+12434000: ticky counter definition 4535063528, arity: 0, def kinds: , name: Main.main1{v r1sI} (fun)
+12434000: ticky counter definition 4535063632, arity: 0, def kinds: , name: Main.main2{v r1sJ} (fun)
+12434000: ticky counter definition 4535063728, arity: 0, def kinds: , name: :Main.main{v 01D} (fun)
```
Comment (maoe) [inline test/ticky-ticky.eventlog.reference:115]: Is it expected that these ref kinds are empty?

Comment (nineonine) [reply test/ticky-ticky.eventlog.reference:115]: I believe so since those closures take no arguments (except go), but @bgamari would know better.

Comment (nineonine) [reply test/ticky-ticky.eventlog.reference:115]: Actually, this program has no `go` function. Sorry for the confusion. If you want, I can update the program to contain ticky events that have values there. Should I do that?

Comment (maoe) [reply test/ticky-ticky.eventlog.reference:115]: Thanks! No I think it's fine as-is.

**Pull request merged**